### PR TITLE
Drop support for ancient upstreams

### PR DESCRIPTION
**Pull request by mattwthompson**
Drop support for
* Python 2
* PyTables 2
* SciPy <0.12
* Pandas <0.14

Also drops the internal `mdtraj.utils.six` module, which I think was just vendored. That's technically an API break, but I don't think there's good reasons for downstream packages to depend on it (both because Python 2/3 compatibility is bad and because this isn't the way to get `six`).

### CODE REVIEW

File: mdtraj/scripts/mdconvert.py [hunk 1]
```diff
@@ -310,7 +308,7 @@ def main(args, verbose=True):
                             elif isinstance(v, md.Topology):
                                 _data[k] = v
                             else:
-                                raise RuntineError()
```
Comment (ctk3b) [inline mdtraj/scripts/mdconvert.py:313]: 😬

File: tests/test_docstrings.py [hunk 2]
```diff
@@ -33,6 +32,7 @@ def module(request):
     return mod
 
 
+@pytest.mark.skip()
```
Comment (ctk3b) [inline tests/test_docstrings.py:35]: Is this test broken?

File: tests/test_nmr.py [hunk 3]
```diff
@@ -35,8 +33,8 @@ def test_ppm(get_fn):
     #       2     ASN   CA    999.000  53.004  51.168  51.802  53.081  54.098  52.820  52.379  51.856  53.034  52.754  54.134  54.222  51.210  52.207  50.824  54.459  53.605  54.211  53.688  52.344  53.004  51.168  51.802  53.081  54.098  52.820  52.379  51.856  53.034  52.754  54.134  54.222  51.210  52.207  50.824  54.459  53.605  54.211  53.688  52.344
 
 
-@pytest.mark.skipif((not find_executable(SHIFTX2)) or (not six.PY2) or (os.environ.get('TRAVIS', '') == 'true'),
-                    reason='SHIFTX2 binary not found or Python 2 not found or runningn on travis.')
+@pytest.mark.skipif((not find_executable(SHIFTX2)) or (os.environ.get('TRAVIS', '') == 'true'),
+                    reason='SHIFTX2 binary not found or runningn on travis.')
```
Comment (ctk3b) [inline tests/test_nmr.py:37]:
```suggestion
                    reason='SHIFTX2 binary not found or running on travis.')
```

**Review by ctk3b — APPROVED** [review]
LGTM % fixing or updating docs on the `test_docstrings` test

**Comment by mattwthompson** [comment]
There are more Python 2 artifacts that could be re-jiggered, but for now this does everything I originally set out and more. Will leave open briefly in case anybody else really likes reviewing semi-automated changelogs

**Comment by sukritsingh** [comment]
Is #1835 a relevant issue to this PR? I linked it here just for tracking purposes but can unlink if it's not relevant/addressed (I haven't checked out the commits yet tbh)

**Comment by mattwthompson** [comment]
Yep, this should resolve that issue

**Comment by a-detiste** [comment]
https://wiki.debian.org/Python3-six-removal

> Also drops the internal mdtraj.utils.six module, which I think was just vendored. That's technically an API break, but I don't think there's good reasons for downstream packages to depend on it (both because Python 2/3 compatibility is bad and because this isn't the way to get six).

You're so right. So many project does that and now feels like they should do it forever

**Pull request merged**